meson.build

```diff
@@ -232,16 +232,14 @@ asm_is_nasm = false
 asm_args = []
 
 # ASM architecture variables
-asm_option = get_option('asm')
-cpu_subfamily = host_machine.cpu_family()
-if cpu_subfamily.startswith('x86')
-    cpu_family = 'x86'
-else
-    cpu_family = cpu_subfamily
-endif
+cpu_family = host_machine.cpu_family()
+asm_option = get_option('asm').require(
+    cpu_family in ['x86', 'x86_64', 'aarch64'],
+    error_message: 'ASM support is not available on ' + cpu_family,
+)
```

Comment:
i’m not fond of this; it requires updating this list _and_ the configuration logic for all future additions (or if it ever happens: removals)

If ASM was explicitly requested on an unsupported arch the warning informing about it should be immediately before the error anyway iinm

Reply:
IME people will hardly ever read error messages, let alone warnings above it. I found this issue specifically because other people did not read those warnings, and when we moved from autotools to meson, asm support had been silently disabled (because the old build defaulted to `--enable-asm` but now the default is `auto`, although that's a separate issue).

I think it's also an easy way to know at a glance what the arch support is. Unfortunately, meson's syntax doesn't (yet?) support a way to make these things declarative / DRY.

Reply:
> IME people will hardly ever read error messages, let alone warnings above it

If someone doesn’t read error messages after configuration failed, the changes to the unread message here won’t help, but our logic gets more complicated and by duplication more prone to future bugs. If the error message is read, the existing message already explicitly points to prior warnings which will be placed immediately above the error so it’d be hard to miss. Sorry, I don’t see a benefit

> because the old build defaulted to --enable-asm but now the default is auto, although that's a separate issue

Was this an issue in the downstream setup, e.g. omitting `-Dasm=enabled` where `--enable-asm` was set previously, or did some behaviour difference between autotools and the meson sneak in?

Reply:
The issue was that we weren't passing `-Dasm=enabled`.

The difference from autotools was that with meson's toolchain support for nasm, `nasm` needs to be specified in the cross-file to be picked up for cross-builds, unlike a regular program  (this will be fixed in the next meson release).

Reply:
I have removed the `.require()` now.